integration/conftest.py

```diff
@@ -401,9 +401,6 @@ def alice(
             reactor, request, temp_dir, introducer_furl, flog_gatherer, "alice",
             web_port="tcp:9980:interface=localhost",
             storage=False,
-            # We're going to kill this ourselves, so no need for finalizer to
```

Comment:
This was probably the key change.

Reply:
Ah.  `TahoeProcess.restart_async` used to lose the new process that it launched so finalization was broken because it would try to kill the old process which no longer existed.  But then at some point we fixed `restart_async` to keep track and now finalization is the correct thing to do.  Great.  I think.